src/store/modalStore.ts

```diff
@@ -0,0 +1,50 @@
+import { create } from 'zustand';
+import WEEK from 'utils/constant/week';
+
+interface ModalStore {
+  categoryState: [string, number];
+  searchStoreState: string,
+  openTimeState: { [key in typeof WEEK[number]]: string | null },
+  closeTimeState: { [key in typeof WEEK[number]]: string | null },
+  storeClosedState: { [key: string]: boolean },
+  setCategoryState: (state: [string, number]) => void,
+  setSearchStoreState: (state: string) => void,
+  setOpenTimeState: (state: { [key in typeof WEEK[number]]: string | null }) => void,
+  setCloseTimeState: (state: { [key in typeof WEEK[number]]: string | null }) => void,
+  setStoreClosedState: (state: { [key : string]: boolean }) => void,
+}
+
+const initialOperatingTime = {
+  월: '00:00',
+  화: '00:00',
+  수: '00:00',
+  목: '00:00',
+  금: '00:00',
+  토: '00:00',
+  일: '00:00',
+};
+
+const initialStoreClosed = {
+  월: false,
+  화: false,
+  수: false,
+  목: false,
+  금: false,
+  토: false,
+  일: false,
+};
```

Comment:
```suggestion
const daysOfWeek = ['월', '화', '수', '목', '금', '토', '일'];

const initialOperatingTime = {};
const initialStoreClosed = {};

daysOfWeek.forEach(day => {
  initialOperatingTime[day] = '00:00';
  initialStoreClosed[day] = false;
});
```

이런식으로 같은 속성의 중복성을 줄일수도 있을것같아요. 

(+ 상수명은 대문자로 맞춰주세요)

Reply:
어아 이부분은 저는 조금 보수적인게, store의 initialValue로 들어가는부분이다보니 export되지 않은 전역변수 부분에서 값을 핸들링하는건 위험하다고 느낄 수 있다고 봐요. (단순한 유틸 파일이 아니다보니 더욱 그런 감이 있네요)

차라리 이 변수들을 사용하는 측면에서 명확한 값을 갖게끔 만들어주는건 어떨까 의견을 내봅니다

```ts
const useModalStore = create<ModalStore>((set) => ({
  categoryState: ['', 0],
  searchStoreState: '',
  openTimeState: daysOfWeek.reduce((acc, cur) => { return cur[acc] = '00:00' }, {}),
  // ...
}))
```

Reply:
다시 보니 구조가 명확히 눈에 보인다는 점에서 기존 구조를 유지하는것도 상당한 장점을 가지는 것 같네요.
reduce나 forEach로 보니 가독성이 떨어지는듯. (이 값이 뭐지...하고)